src/inference/smc.js

```diff
@@ -71,6 +79,20 @@ module.exports = function(env) {
     return this.runCurrentParticle();
   };
 
+  // Error function for error handling
+  // this.throwOnError is true: directly throw error
+  // this.throwOnError is false: return error (string) as infer result
+  SMC.prototype.error = function(errType) {
+    if (this.throwOnError) {
+      throw new Error(errType);
+    } else {
+      this.err = errType;
+      // the way to return differs from enumerate and rejection because of CPS
+      // directly calling this.k doesn't work
+      return this.finish();
```

Comment:
I think it would be nice if these `error` methods worked in the same way. What doesn't work when calling `this.k` directly here? (I've tried to figure it out be can't see it.)